libgnucash/engine/Transaction.c

```diff
@@ -665,6 +665,10 @@ xaccTransCloneNoKvp (const Transaction *from)
 			    qof_instance_get_book(from));
 
     xaccTransBeginEdit(to);
+
+    xaccTransSetNotes(to, xaccTransGetNotes(from));
+    xaccTransSetDocLink(to, xaccTransGetDocLink(from));
+
```

Comment:
Notes and DocLink are stored in KVP, so this breaks the documented purpose of the function.

Reply:
Good point.  I've moved the change into xaccTransClone().

Thanks.

Reply:
Where it shouldn't be necessary, `xaccTransClone` calls `qof_instance_copy_kvp` that should already copy the notes, doclink, and everything else stored in KVP.

`xaccTransReverse` calls `xaccTransClone`, meaning that the intended behavior is what you want and the discussion about use-case is irrelevant.

Testing shows that it does, that the doclink mark shows on the reversed transaction and Manage Document Links shows the same link as the original. *But* although the notes slot is duplicated in the reversed transaction it doesn't display in the register until one closes and reopens the book; simply closing and reopening the register isn't sufficient. It's pretty odd because the doclink code is identical both in Transaction.c and split-register-model.c.